python/lsst/sconsUtils/tests.py

```diff
@@ -293,3 +305,38 @@ def junitPrefix(self):
             prefix += ".{0}".format(os.environ[controlVar])
 
         return prefix
+
+    def _pytestCoverage(self, prefix="pytest"):
+        """Form the additional arguments required to enable coverage testing.
+        Prefix is used to form the output files.
+        """
+
+        options = ""
+
+        # if we have a standard package layout, restrict coverage to the
+        # python code and do not include the tests.
+        if os.path.exists("python") or os.path.exists(os.path.join(os.path.pardir, "python")):
+            options += " --cov=python"
+        else:
+            # Default to coverage for everything in this directory and below
+            options += " --cov=."
+
+        options += " --cov-branch --cov-report=term "
+
+        covfile = "{}-cov-{}.xml".format(prefix, self._env['eupsProduct'])
+
+        # We should specify the output directory explicitly unless the prefix
+        # indicates that we are using the SCons target
+        if covfile.startswith("${TARGET}"):
+            covpath = covfile
+        else:
+            covpath = os.path.join(self._tmpDirAbs, covfile)
+        options += " --cov-report=xml:'{}'".format(covpath)
+
+        # Use a prefix for the HTML if the prefix is not the default
+        htmlfile = ""
+        if prefix != "pytest":
+            htmlfile = ":'htmlcov-{}'".format(prefix)
```

Comment:
This puts the output in the package root directory: `./htmlcov`: can we get it put into the `tests/` directory, or maybe in `doc/`? The xml report is written to `tests/.tests/pytest-cov-*.xml`, so `tests/.tests/htmlcov` seems like a good place.

We also need to add it to the `.gitignore` and as a `scons --clean` target.